tests/fpgadataflow/test_convert_to_hw_thresholding.py

```diff
@@ -162,7 +177,31 @@ def test_convert_multithreshold_to_hardware(
     )
 
     model = model.transform(InferThresholdingLayer())
+
+    # Perform functional validation of the InferThresholdingLayer transform
+    x = gen_finn_dt_tensor(input_data_type, tuple(num_input_vecs + [num_input_channels]))
+
+    x_nchw = layout_FINN2NCHW(x)
+    y_expected = multithreshold(x_nchw, thresholds)
+
+    # convert back to NHWC for comparison to hw outputs
+    y_expected = layout_NCHW2FINN(y_expected)
+    if activation == DataType["BIPOLAR"]:
```

Comment:
Above, you are skipping if the activation is `DataType["BIPOLAR"]` with the reasoning that it doesn't allow for negative values. This is not correct, the `.signed()` method checks if the minimal allowed value is below zero: https://github.com/fastmachinelearning/qonnx/blob/main/src/qonnx/core/datatype.py#L38
For bipolar, that is -1: https://github.com/fastmachinelearning/qonnx/blob/main/src/qonnx/core/datatype.py#L241